src/libs/DateUtils.js

```diff
@@ -10,10 +10,9 @@ let timezone;
 Onyx.connect({
     key: ONYXKEYS.MY_PERSONAL_DETAILS,
     callback: (val) => {
-        if (val) {
-            timezone = _.isObject(val) ? val.timezone.selected : val.timezone;
-        } else {
-            timezone = CONST.DEFAULT_TIME_ZONE.selected;
+        timezone = val ? val.timezone : CONST.DEFAULT_TIME_ZONE.selected;
```

Comment:
Hi Nikki, would you mind adding some code comments to explain why the logic needs to be this way? Just looking at the code it is very confusing and could use some context. Also, I am curious if lodash.get() would be helpful at all?

Reply:
Sure, I can add some comments here

Reply:
Lodash get could also be helpful here for sure, though I think using a ternary accomplishes the same logic in the same number of lines - open to modifying it though!

Reply:
I think the part that is confusing when I look at the code is that there is a ternary followed by an `if`, and I can't really reason about why that is necessary. Maybe if I had some more context around why the logic is this way, I could provide some more helpful suggestions.

Reply:
For some reason, I can't attach 👍  to your comment but I see your point, and a clarifying comment would definitely be helpful here